test(wpt): Enable wpt workers tests

yacinehmito opened the pull request:
In Web Crypto, must test suites have their JavaScript entrypoint suffixed with `.worker.js`. These files were by default ignored by Deno's integration tests; now they are included.

JavaScript entrypoints meant to be included in HTML files reference their dependencies with `META` comments. JavaScript entrypoints for workers don't do that: they call a function `importScripts()`.
We therefore had to look for the patterns `importScripts("<path>");` in order to build the bundle for the test, as well as included in the bundle the function `importScripts()` as a no-op.

cc @lucacasonato

<!--
Before submitting a PR, please read
https://github.com/denoland/deno/blob/master/docs/contributing.md

1. Give the PR a descriptive title.

  Examples of good title:
    - fix(std/http): Fix race condition in server
    - docs(console): Update docstrings
    - feat(doc): Handle nested reexports

  Examples of bad title:
    - fix #7123
    - update docs
    - fix bugs

2. Ensure there is a related issue and it is referenced in the PR text.
3. Ensure there are tests that cover the changes.
4. Ensure `cargo test` passes.
5. Ensure `./tools/format.js` passes without changing files.
6. Ensure `./tools/lint.js` passes.
-->

lucacasonato commented on cli/tests/integration_tests.rs:
```diff
@@ -5082,6 +5082,8 @@ fn concat_bundle(
     bundle_line_count += 1;
   }
 
+  bundle.push_str("function importScripts() {}\n");
```
```suggestion
  bundle.push_str("function importScripts() {}\n");
  bundle_line_count += 1;
```

source mapping wont be correct otherwise.

yacinehmito replied:
The linter is warning me that `bundle_line_count` is not read after that line.
I'm putting the function declaration at the very top of the bundle then.

lucacasonato reviewed (commented):
`importScripts()` is actually web standard. it is a function that is used to import things in classic mode workers (which we dont support). This is pretty hacky, but I guess it will work for now.

yacinehmito commented:
> `importScripts()` is actually web standard. it is a function that is used to import things in classic mode workers (which we dont support). This is pretty hacky, but I guess it will work for now.

I will add a comment in the code stating that, and will open an issue for the lack of `importScripts()` if I don't find an existing one.

lucacasonato commented:
You can open an issue, but we won't add support for "classic" workers. We are ESM only. Deno will only support "module" type workers.

caspervonb commented:
Looking at the actual tests, e.g digest https://github.com/web-platform-tests/wpt/tree/a51d36327febebc8634d7b523a7e620f4ebdbf26/WebCryptoAPI/digest

Looks like this can be fixed upstream to be an "any" test instead.

All the variants do is import the shared script and run it.

yacinehmito commented:
Oh I see. I won't bother then. Thanks for the clarification.

yacinehmito commented:
@caspervonb I think we might have a hard time pushing changes upstream in timely manner. There are no owners at the moment for the Web Crypto API tests. Also, my guess is that it is suffixed by `worker.js` because it only works in a worker context (which is indeed the case per the call of `importScripts()`).

caspervonb commented:
> @caspervonb I think we might have a hard time pushing changes upstream in timely manner. There are no owners at the moment for the Web Crypto API tests. Also, my guess is that it is suffixed by `worker.js` because it only works in a worker context (which is indeed the case per the call of `importScripts()`).

Hmm not sure about that but I opened two patches yesterday and both got merged within a couple of hours. But we won't know unless we try 🙃

yacinehmito commented:
> Hmm not sure about that, I opened two patches yesterday and both got merged within a couple of hours. 

I meant on the Web Crypto API tests specifically.
They are looking for someone: https://github.com/web-platform-tests/wpt/issues/27111
And this PR of mine is still hanging (although it's not very old): https://github.com/web-platform-tests/wpt/pull/27105

yacinehmito commented:
New changes. Everything should be good now:
- Added comments
- Moved `importScripts` no-op at the beginning of the bundle
- Ignoring imports of `testharness.js`, as it seems to override some important functions, leading to the removal of the test report in the terminal.

lucacasonato reviewed (approved):
LGTM. Thanks
